Source/AwsCommonRuntimeKit/mqtt/Mqtt5Client.swift

```diff
@@ -121,44 +121,154 @@ public typealias OnWebSocketHandshakeInterceptComplete = (HTTPRequestBase, Int32
 /// such as signing/authorization etc... Returning from this function does not continue the websocket
 /// handshake since some work flows may be asynchronous. To accommodate that, onComplete must be invoked upon
 /// completion of the signing process.
-public typealias OnWebSocketHandshakeIntercept = (HTTPRequest, @escaping OnWebSocketHandshakeInterceptComplete) -> Void
+public typealias OnWebSocketHandshakeIntercept = (HTTPRequest, @escaping OnWebSocketHandshakeInterceptComplete) async -> Void
 
 // MARK: - Mqtt5 Client
 public class Mqtt5Client {
-    private var rawValue: UnsafeMutablePointer<aws_mqtt5_client>?
-    private var callbackCore: MqttCallbackCore
+    internal var clientCore: Mqtt5ClientCore
 
+    /// Creates a Mqtt5Client instance using the provided MqttClientOptions. It is MANDATORY to call `close()`
+    /// to clean up the Mqtt5Client resource
+    ///
+    /// - Parameters:
+    ///     clientOptions: The MqttClientOptions class to use to configure the new Mqtt5Client.
+    ///
+    /// - Throws: CommonRuntimeError.crtError If the system is unable to allocate space for a native MQTT5 client structure
+    init(clientOptions options: MqttClientOptions) throws {
+        try options.validateConversionToNative()
```

Comment:
Maybe not part of this PR but this is redundant, the core also validates it. We should just update CStruct to update or add more functions which do the validation. I think we should just get rid of all the validateConversion* functions. It should be part of CStruct interface, so that we can have a consistent way of doing this.

Reply:
The change might introduce changes in CStruct interface. I will think about it and update in a later PR.